crates/ruff_linter/src/rules/flake8_type_checking/helpers.rs

```diff
@@ -269,6 +269,20 @@ pub(crate) fn quote_annotation(
         }
     }
 
+    quote_type_expression(expr, semantic, stylist)
+}
+
+/// Wrap a type expression in quotes.
+///
+/// This function assumes that the callee already expanded expression components
+/// to the minimum acceptable range for quoting, i.e. the parent node may not be
+/// a [`Expr::Subscript`], [`Expr::Attribute`], `[Expr::Call]` or `[Expr::BinOp]`.
+pub(crate) fn quote_type_expression(
```

Comment:
Is this what `quote_annotation` does? If so, it might be worth to link directly to the function